core/backend/src/IModelDb.ts

```diff
@@ -2528,7 +2529,9 @@ export class SnapshotDb extends IModelDb {
    */
   public static async openCheckpointV2(checkpoint: CheckpointProps): Promise<SnapshotDb> {
     const { filePath, expiryTimestamp } = await V2CheckpointManager.attach(checkpoint);
-    const snapshot = SnapshotDb.openFile(filePath, { lazyBlockCache: true, key: CheckpointManager.getKey(checkpoint) });
+    const key = CheckpointManager.getKey(checkpoint);
+    const tempFileBase = join(IModelHost.cacheDir, key); // temp files for this checkpoint should go in the cacheDir.

```

Comment:
Does something ensure the `key` subdirectory exists?

Reply:
it seems to me that `IModelHost.setupCacheDirs` should do that, but it doesn't look to me like it does (unless I'm missing something.)

Reply:
but, tempFileBase is a filename, not a directory name (in this case it's a file that doesn't exist).

Reply:
Do you intend to fix setupCacheDirs in this PR?

Reply:
it actually already happens [here](https://github.com/imodeljs/imodeljs/blob/2592196e66439a657cbbe9ef5d5a3673c15d4307/core/backend/src/BriefcaseManager.ts#L76-L79) although that seems a bit obscure to me.

Reply:
i added a direct call in `IModelHost.setupCacheDirs`